File: FrontEnd (C#)/SoftProgPersistance/GestMaterial/Impl/BibliotecaImpl.cs
using SoftProgDBManager;
using SoftProgModel.GestMaterial;
using SoftProgPersistance.GestMaterial.DAO;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgPersistance.GestMaterial.Impl
{
    public class BibliotecaImpl : BibliotecaDAO
    {
        private DbDataReader lector;
        public int eliminar(int idObjeto)
        {
            DbParameter[] parametros = new DbParameter[1];
            parametros[0] = DBManager.Instance.CreateParam("_id_biblioteca", DbType.Int32, idObjeto, ParameterDirection.Input);
            return DBManager.Instance.EjecutarProcedimiento("ELIMINAR_BIBLIOTECA", parametros);
        }

        public int insertar(Biblioteca biblioteca)
        {
            DbParameter[] parametros = new DbParameter[3];
            parametros[0] = DBManager.Instance.CreateParam("_id_biblioteca", DbType.Int32, null, ParameterDirection.Output);
            parametros[1] = DBManager.Instance.CreateParam("_nombre", DbType.String, biblioteca.Nombre, ParameterDirection.Input);
            parametros[2] = DBManager.Instance.CreateParam("_ubicacion", DbType.String, biblioteca.Ubicacion, ParameterDirection.Input);
            DBManager.Instance.EjecutarProcedimiento("INSERTAR_BIBLIOTECA", parametros);
            biblioteca.IdBiblioteca = Convert.ToInt32(parametros[0].Value);
            return biblioteca.IdBiblioteca;
        }

        public BindingList<Biblioteca> listarTodos()
        {
            BindingList<Biblioteca> bibliotecas = null;
            lector = DBManager.Instance.EjecutarProcedimientoLectura("LISTAR_BIBLIOTECAS_TODAS", null);
            while (lector.Read())
            {
                if (bibliotecas == null) bibliotecas = new BindingList<Biblioteca>();
                Biblioteca biblioteca = new Biblioteca();
                if (!lector.IsDBNull(lector.GetOrdinal("id_biblioteca"))) biblioteca.IdBiblioteca = lector.GetInt32(lector.GetOrdinal("id_biblioteca"));// Se coloca el identificador del select
                if (!lector.IsDBNull(lector.GetOrdinal("nombre"))) biblioteca.Nombre = lector.GetString(lector.GetOrdinal("nombre"));
                if (!lector.IsDBNull(lector.GetOrdinal("ubicacion"))) biblioteca.Ubicacion = lector.GetString(lector.GetOrdinal("ubicacion"));
                if (!lector.IsDBNull(lector.GetOrdinal("activo"))) biblioteca.Activo = lector.GetBoolean(lector.GetOrdinal("activo"));
                bibliotecas.Add(biblioteca);
            }
            DBManager.Instance.CerrarConexion();
            return bibliotecas;
        }

        public int modificar(Biblioteca biblioteca)
        {
            DbParameter[] parametros = new DbParameter[3];
            parametros[0] = DBManager.Instance.CreateParam("_id_biblioteca", DbType.Int32, null, ParameterDirection.Output);
            parametros[1] = DBManager.Instance.CreateParam("_nombre", DbType.String, biblioteca.Nombre, ParameterDirection.Input);
            parametros[2] = DBManager.Instance.CreateParam("_ubicacion", DbType.String, biblioteca.Ubicacion, ParameterDirection.Input);
            return DBManager.Instance.EjecutarProcedimiento("MODIFICAR_BIBLIOTECA", parametros);
        }

        public Biblioteca obtenerPorId(int idObjeto)
        {
            Biblioteca biblioteca = null;
            DbParameter[] parametros = new DbParameter[1];
            parametros[0] = DBManager.Instance.CreateParam("id_biblioteca", DbType.Int32, idObjeto, ParameterDirection.Input);
            lector = DBManager.Instance.EjecutarProcedimientoLectura("OBTENER_BIBLIOTECA_X_ID", parametros);
            if (lector.Read())
            {
                if (biblioteca == null) biblioteca = new Biblioteca();
                if (!lector.IsDBNull(lector.GetOrdinal("id_biblioteca"))) biblioteca.IdBiblioteca = lector.GetInt32(lector.GetOrdinal("id_biblioteca"));// Se coloca el identificador del select
                if (!lector.IsDBNull(lector.GetOrdinal("nombre"))) biblioteca.Nombre = lector.GetString(lector.GetOrdinal("nombre"));
                if (!lector.IsDBNull(lector.GetOrdinal("ubicacion"))) biblioteca.Ubicacion = lector.GetString(lector.GetOrdinal("ubicacion"));
                if (!lector.IsDBNull(lector.GetOrdinal("activo"))) biblioteca.Activo = lector.GetBoolean(lector.GetOrdinal("activo"));
            }
            DBManager.Instance.CerrarConexion();
            return biblioteca;
        }

        public BindingList<Biblioteca> listar_bibliotecas_por_material(int _id_material)
        {
            BindingList<Biblioteca> bibliotecas = null;
            DbParameter[] parametros = new DbParameter[1];
            parametros[0] = DBManager.Instance.CreateParam("_id_material", DbType.Int32, _id_material, ParameterDirection.Input);

            lector = DBManager.Instance.EjecutarProcedimientoLectura("LISTAR_BIBLIOTECAS_DE_MATERIAL", parametros);
            while (lector.Read())
            {
                if (bibliotecas == null) bibliotecas = new BindingList<Biblioteca>();
                Biblioteca biblioteca = new Biblioteca();
                if (!lector.IsDBNull(lector.GetOrdinal("nombre"))) biblioteca.Nombre = lector.GetString(lector.GetOrdinal("nombre"));
                bibliotecas.Add(biblioteca);
            }
            DBManager.Instance.CerrarConexion();



            return bibliotecas;
        }
    }
}

File: BackEnd (Java)/BibliotecaDA/src/main/java/biblioteca/gestionMaterial/mysql/BibliotecaImpl.java

package biblioteca.gestionMaterial.mysql;

import biblioteca.config.DBManager;
import biblioteca.gestionMaterial.dao.BibliotecaDAO;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Types;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;
import pe.edu.pucp.utilsarmy.gestion_de_material.model.Biblioteca;



public class BibliotecaImpl implements BibliotecaDAO{
    
    private ResultSet rs;
    
    @Override
    public int insertar(Biblioteca objeto) {
        Map<Integer,Object> parametrosSalida = new HashMap<>();
        Map<Integer,Object> parametrosEntrada = new HashMap<>();
        parametrosSalida.put(1, Types.INTEGER);
        parametrosEntrada.put(2, objeto.getNombre());
        parametrosEntrada.put(3, objeto.getUbicacion());
        parametrosEntrada.put(4, objeto.isActivo());
        DBManager.getInstance().ejecutarProcedimiento("INSERTAR_BIBLIOTECA", parametrosEntrada, parametrosSalida);
        objeto.setIdBiblioteca((int) parametrosSalida.get(1));
        System.out.println("Se ha realizado el registro de la biblioteca");
        return objeto.getIdBiblioteca(); 
    }
    

    @Override
    public int modificar(Biblioteca objeto) {
        Map<Integer, Object> parametrosEntrada = new HashMap<>();
        parametrosEntrada.put(1, objeto.getIdBiblioteca());
        parametrosEntrada.put(2, objeto.getNombre());
        parametrosEntrada.put(3, objeto.getUbicacion());
        parametrosEntrada.put(4, objeto.isActivo());
        int resultado = DBManager.getInstance().ejecutarProcedimiento("MODIFICAR_BIBLIOTECA", parametrosEntrada, null);
        System.out.println("Se ha realizado la modificacion de la biblioteca");
        return resultado;
    }

    @Override
    public int eliminar(int idObjeto) {
        Map<Integer, Object> parametrosEntrada = new HashMap<>();
        parametrosEntrada.put(1, idObjeto);
        int resultado = DBManager.getInstance().ejecutarProcedimiento("ELIMINAR_BIBLIOTECA", parametrosEntrada, null);
        System.out.println("Se ha realizado la eliminacion de la biblioteca");
        return resultado;
    }

    @Override
    public Biblioteca obtenerPorId(int idObjeto) {
        Biblioteca biblioteca = null;
        Map<Integer, Object> parametrosEntrada = new HashMap<>();
        parametrosEntrada.put(1, idObjeto);
        rs = DBManager.getInstance().ejecutarProcedimientoLectura("OBTENER_BIBLIOTECA_X_ID", parametrosEntrada);
        System.out.println("Lectura de biblioteca...");
        try{
            if(rs.next()){
                biblioteca = new Biblioteca();
                biblioteca.setIdBiblioteca(rs.getInt("id_biblioteca"));
                biblioteca.setNombre(rs.getString("nombre"));
                biblioteca.setUbicacion(rs.getString("ubicacion"));
                biblioteca.setActivo(rs.getBoolean("activo"));
                
            }
        }catch(SQLException ex){
            System.out.println("ERROR: " + ex.getMessage());
        }finally{
            DBManager.getInstance().cerrarConexion();
        }
        return biblioteca;
    }
    
    @Override
    public ArrayList<Biblioteca> listarTodos() {
        ArrayList<Biblioteca> bibliotecas = null;
        rs = DBManager.getInstance().ejecutarProcedimientoLectura("LISTAR_BIBLIOTECAS_TODAS", null);
        System.out.println("Lectura de todas las bibliotecas...");
        try{
            while(rs.next()){
                if(bibliotecas == null) bibliotecas = new ArrayList<>();
                Biblioteca bio = new Biblioteca();
                bio.setIdBiblioteca(rs.getInt("id_biblioteca"));
                bio.setNombre(rs.getString("nombre"));
                bio.setUbicacion(rs.getString("ubicacion"));
                bio.setActivo(rs.getBoolean("activo"));
                bibliotecas.add(bio);
            }
        }catch(SQLException ex){
            System.out.println("ERROR: " + ex.getMessage());
        }finally{
            DBManager.getInstance().cerrarConexion();
        }
        return bibliotecas;
    }
    
}
